backend/src/parser/TheParser.ts

```diff
@@ -441,6 +441,10 @@ export default class TheParser {
     }
 
     validUrl(url: string) {
-        return encodeURI(decodeURI(url)).match(urlRegexExact);
+        try {
+            return encodeURI(decodeURI(url)).match(urlRegexExact);
```

Comment:
Let's made method typing more strict - `match` returns array and downstream methods may be expecting it.
```suggestion
            return encodeURI(decodeURI(url)).match(urlRegexExact) !== null;
```